### PR TITLE
geojson directive control & geojsonWatchOptions

### PR DESCRIPTION
**Pull request by nmccready**
- DirectiveControls specs
- geojsonWatchOptions directive
- directivecontrols extended to geojson

### CODE REVIEW

File: src/services/leafletWatchHelpers.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
+angular.module("leaflet-directive")
```
Comment (nmccready) [inline src/services/leafletWatchHelpers.js]: moving watch code boilerplate to a service

File: src/services/leafletMarkersHelpers.js [hunk 2]
```diff
@@ -475,29 +475,6 @@ angular.module("leaflet-directive")
             }, isDeepWatch);
         },
```
Comment (nmccready) [inline src/services/leafletMarkersHelpers.js]: moved to leafletHelpers to be generic

File: src/services/leafletDirectiveControlsHelpers.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        }
+        else if(_isObject(thingToAddName) && !_isDefined(createFn) && !_isDefined(cleanFn)){
+            extender = thingToAddName;
+        }
+        else{
+            $log.error(_fnHeader + 'incorrect arguments');
+            return;
+        }
+
+        //add external control to create / destroy markers without a watch
```
Comment (nmccready) [inline src/services/leafletDirectiveControlsHelpers.js]: main logic here where the controls is extended for the intended directive

File: src/services/leafletDirectiveControlsHelpers.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    var _isDefined = leafletHelpers.isDefined,
+        _isString = leafletHelpers.isString,
+        _isObject = leafletHelpers.isObject,
+        _mainErrorHeader = leafletHelpers.errorHeader;
+
+    var _errorHeader = _mainErrorHeader + '[leafletDirectiveControlsHelpers';
+
+    var _extend = function(id, thingToAddName, createFn, cleanFn){
+        var _fnHeader = _errorHeader + '.extend] ';
+        var extender = {};
```
Comment (nmccready) [inline src/services/leafletDirectiveControlsHelpers.js]: bunch of error handling for different argument usecases

File: src/directives/markersWatchOptions.js [hunk 5]
```diff
@@ -1,31 +0,0 @@
-angular.module("leaflet-directive").directive('markersWatchOptions',
```
Comment (nmccready) [inline src/directives/markersWatchOptions.js]: removed in favor of watchOptions.js which creates multiple directive control directives to handle common defaults.

File: src/directives/markers.js [hunk 6]
```diff
@@ -182,19 +177,10 @@ angular.module("leaflet-directive").directive('markers',
                         _addMarkers(models, map, layers, leafletMarkers, leafletScope,
                             watchOptions);
                     };
```
Comment (nmccready) [inline src/directives/markers.js]: removed boilerplate in favor of services

File: src/directives/geojson.js [hunk 7]
```diff
@@ -1,4 +1,11 @@
-angular.module("leaflet-directive").directive('geojson', function ($log, $rootScope, leafletData, leafletHelpers) {
+angular.module("leaflet-directive")
```
Comment (nmccready) [inline src/directives/geojson.js]: now supports directiveControls and watchOptions for geojson

**Comment by nmccready** [comment]
@tombatossals @zacronos

**Comment by zacronos** [comment]
LGTM

**Pull request merged**